### PR TITLE
NEW: Added canRunInBackground flag in device debug view

### PR DESCRIPTION
**Pull request by bmalrat**
### Description

When investigation this issue https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-415 it was not possible to know that some device can run in back ground or not.

### Changes made

Added an internal editor only property to be able to access to device flags without the editor override that would hide the device capability.
Screenshot 2024-09-25 131602

### Testing

Tested locally on windows with ISXB-415 project + auto test

### Risk

no change, only new internal api

### Checklist

Before review:

- [X] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - JIRA ticket linked, example ([case %<ID>%](https://issuetracker.unity3d.com/product/unity/issues/guid/<ID>)). If it is a private issue, just add the case ID without a link.
    - Jira port for the next release set as "Resolved".
- [X ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

After merge:

- [ ] Create forward/backward port if needed. If you are blocked from creating a forward port now please add a task to ISX-1444.

### CODE REVIEW

File: Packages/com.unity.inputsystem/InputSystem/InputManager.cs [hunk 1]
```diff
@@ -2893,22 +2893,8 @@ internal void AddAvailableDevicesThatAreNowRecognized()
 
         private bool ShouldRunDeviceInBackground(InputDevice device)
         {
-            var runDeviceInBackground =
-                m_Settings.backgroundBehavior != InputSettings.BackgroundBehavior.ResetAndDisableAllDevices &&
+            return m_Settings.backgroundBehavior != InputSettings.BackgroundBehavior.ResetAndDisableAllDevices &&
                 device.canRunInBackground;
-
-            // In editor, we may override canRunInBackground depending on the gameViewFocus setting.
-            #if UNITY_EDITOR
```
Comment (bmalrat) [inline Packages/com.unity.inputsystem/InputSystem/InputManager.cs:2901]: this code is already done in canRunInBackground

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,8 @@ however, it has to be formatted properly to pass verification tests.
 - Renamed editor Resources directories to PackageResources to fix package validation warnings.
 - Changed representation of GamepadButton enum values in Inspector to display aliased enum values as a single item to avoid confusion around selection and aliased value display when multiple enum items map to the same numerical value. [ISXB-543](https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-543)
 - PlayerInput component now warns if the system cannot find matching control scheme, which can occur if all control schemes already paired (e.g. to other game objects with PlayerInput components) [ISXB-1020](https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-1020)
+### Added
+- Added the display of the device flag CanRunInBackground in device debug view.
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: Minor: Code format `CanRunInBackground`

**Review by ekcoh — APPROVED** [review]
LGTM, good observability improvement.

File: Packages/com.unity.inputsystem/InputSystem/Devices/InputDevice.cs [hunk 3]
```diff
@@ -247,14 +247,27 @@ public bool canRunInBackground
                 // In the editor, "background" refers to "game view not focused", not to the editor not being active.
                 // So, we modulate canRunInBackground depending on how input should behave WRT game view according
                 // to the input settings.
-                #if UNITY_EDITOR
+#if UNITY_EDITOR
```
Comment (lyndon-unity) [inline Packages/com.unity.inputsystem/InputSystem/Devices/InputDevice.cs]: Personally I find this multi function spanning #if a bit hard to parse. Would love it if this could be simplified.

**Review by lyndon-unity — APPROVED** [review]

File: Packages/com.unity.inputsystem/InputSystem/Devices/InputDevice.cs (hunk 3 above)
Comment (jfreire-unity) [reply Packages/com.unity.inputsystem/InputSystem/Devices/InputDevice.cs]: Agree. From a readability perspective, it's not great. Can this be improved @bmalrat ?

**Review by jfreire-unity — APPROVED** [review]
